### DISCUSSION TITLE
TemplatedControl Bindings not working

### DISCUSSION THREAD

**Discussion by prismamed** [AvaloniaUI/Avalonia#11759] [Q&A] [answered]
Hello. I'm new to Avalonia.

AvaloniaUi 11-rc1.1
I'm struggling with a Templated Control and Bindings.

Creating a Templated Control with a StyledProperty is easy and I understand this. Here is an example (using the Template Avalonia provided).

This is my TestControl for understanding

**XAML**
```
<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        x:DataType="controls:TestControl"
        xmlns:controls="using:AvaloniaApplication6">
    <Design.PreviewWith>
        <StackPanel>
            <TextBox Name="demoBox">Hallo Welt</TextBox>
            <controls:TestControl Text="{Binding #demoBox.Text}" />
        </StackPanel>
    </Design.PreviewWith>

    <Style Selector="controls|TestControl">
        <!-- Set Defaults -->
        <Setter Property="Template">
            <ControlTemplate>
                <TextBlock Text="{TemplateBinding Text}" />
            </ControlTemplate>
        </Setter>
    </Style>
</Styles>
```

Code Behind

```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Controls.Primitives;
using Avalonia.Data;

namespace AvaloniaApplication6;

public class TestControl : TemplatedControl
{
    public static readonly StyledProperty<string> TextProperty = AvaloniaProperty.Register<TestControl, string>(
        "Text", "nix", false, BindingMode.TwoWay);

    public string Text
    {
        get => GetValue(TextProperty);
        set => SetValue(TextProperty, value);
    }
}
```

Now I'm trying to use this control. I want to bind to my Text Property.

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:avaloniaApplication6="clr-namespace:AvaloniaApplication6"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication6.MainWindow"
        Title="AvaloniaApplication6">
    <Border Padding="20">
        <StackPanel Spacing="20">
            <TextBox Name="demoBox">Hello World</TextBox>
            <avaloniaApplication6:TestControl Text="{Binding #demoBox.Text }" />
            <TextBlock Text="{Binding #demoBox.Text }"></TextBlock>
        </StackPanel>
    </Border>
</Window>
```

But the Control shows simply nothing.
The TextBlock below shows the value of the TextBox.

So could please tell me someone where's the catch?

Thank you very much.

**Comment by prismamed** [comment, marked as answer]
So suddenly I found a solution.
It's really bad that the documentation lacks this information and I have wade through a lot discussions to find eventually an answer.

The answer is:

Include the control into the Application.Styles like this.

```
<StyleInclude Source = "avares://AvaloniaApplication6/TestControl.axaml" />
```

**Reply by timunie** [reply]
we have a detailed sample in Avalonia.Samples. Moreover we are preparing new docs for 11.0 release. So stay tuned. If then anything is still missing from the docs, feel free to help us by providing a PR (But again, only after 11.0 docs are there).

**Comment by prismamed** [comment]
Hi Tim,

Thank you very much.
We really want to invest into AvaloniaUi (looking at XPF :). I think it’s the best Xplat Framework available.

I’m looking forward for the new docs.

Sincerly

Viele Grüße
Daniel Hüttenberger

Prismamed GmbH
Franz-Joseph-Str. 38
80801 München
T 089 21 52 78 69
E ***@***.***
